### ISSUE TITLE
s2forums archiver implementation

### ISSUE THREAD

**Issue by lankybox02** [ScratchAddons/ScratchAddons#6411]
### Why this would be helpful

This site: https://s2forums-archiver.glitch.me/
is pretty dang useful. I think it would be pretty cool to have a button that I can just click and the entire forum topic is downloaded.

### How this addon works

It adds a button to the first page (only the first page) of forum topics. When clicked, it fetches `https://s2forums-archiver.glitch.me/[topic id]` and saves the result.

**Comment by rgantzos** [comment]
In what scenarios would a person want to download a forum post? It's accessible unless it's taken down, in which case (if for some reason you needed to view it) you could do so on Ocular.

**Comment by CST1229** [comment]
> In what scenarios would a person want to download a forum post? It's accessible unless it's taken down, in which case (if for some reason you needed to view it) you could do so on Ocular.

I think ocular no longer shows deleted posts (but you can view ScratchDB's API directly instead).

**Comment by rgantzos** [comment]
> > In what scenarios would a person want to download a forum post? It's accessible unless it's taken down, in which case (if for some reason you needed to view it) you could do so on Ocular.
> 
> I think ocular no longer shows deleted posts (but you can view ScratchDB's API directly instead).

I tested it and it does still include deleted posts.

**Comment by lankybox02** [comment]
> It would be ideal if s2forums-archiver provided a bookmarklet that did this.

`id=prompt('topicID');window.open(`//s2forums-archiver/`+id);`

> In what scenarios would a person want to download a forum post? It's accessible unless it's taken down, in which case (if for some reason you needed to view it) you could do so on Ocular.

Sure, you can use Ocular, but s2archiver saves topics in JSON format, which is great for things like archiving in mass or just downloading topics directly on your computer instead of trusing a centralised service (scratchDB) to show you deleted posts.

**Comment by lankybox02** [comment]
Suggestion: Maybe a button like this could be added
![image](https://github.com/ScratchAddons/ScratchAddons/assets/104015308/2bb4b6c7-5dc5-4392-9281-c6c227c47ecc)
I've been working on something [here](https://github.com/lankybox02/ScratchAddons/) but it has errors

**Comment by BroJac5246** [comment]
What is the use case? Can't you just use Ocular or even Ctrl+S?

**Comment by WorldLanguages** [comment]
> NOTE: This archiver only saves the first page of the topic (aproximately ~20 posts).

Yeah, as long as this is the case, Ctrl+S does the same thing as "archiving"

**Comment by WorldLanguages** [comment]
I think we could have an addon that archives a full forum topic (possibly up to 25 pages or so) but it can be done client-side, no need for a Glitch server

**Comment by rgantzos** [comment]
> I think we could have an addon that archives a full forum topic (possibly up to 25 pages or so) but it can be done client-side, no need for a Glitch server

That would be possible, but what's the use case, especially if it can be done without an addon? This seems extremely technical. You can use Ocular if for some reason a post gets taken down that you find the need to save. There are similar websites such as the Internet Archive, if you want to do it manually.

**Comment by WorldLanguages** [comment]
> That would be possible, but what's the use case

Frequent users of the forums will find use-cases for anything, trust me.

**Comment by BroJac5246** [comment]
> > That would be possible, but what's the use case
> 
> Frequent users of the forums will find use-cases for anything, trust me.
> 

Sure, but would it actually be enough to justify this? Particularly considering the alternatives? And if you need super badly for some unforseen reason, you can use the bookmarklet. Also, this relies on a third party. Do we want that?

**Comment by WorldLanguages** [comment]
@BroJac5246 It wouldn't be hard to implement this in a way that works client-side only.

**Comment by lankybox02** [comment]
You could make use of the code: https://glitch.com/~s2forums-archiver

**Comment by Samq64** [comment]
The site is no longer up.
